src/analysis/retail/evoker/preservation/normalizers/CastLinkNormalizer.ts

```diff
@@ -83,6 +83,9 @@ const EVENT_LINKS: EventLink[] = [
         c.hasTalent(TALENTS_EVOKER.RESONATING_SPHERE_TALENT)
       );
     },
+    additionalCondition(linkedEvent, referencedEvent) {
+      return !HasRelatedEvent(linkedEvent, FROM_HARDCAST);
+    },
```

Comment:
is the order of link processing deterministic based on the order in which links are provided? @kfinch

Reply:
It always links in order, so rules above the current rule will already have been linked when considering the current link.